apollo-router/src/plugins/telemetry/config_new/events.rs

```diff
@@ -575,66 +513,79 @@ struct SubgraphEventsConfig {
 #[derive(Deserialize, JsonSchema, Clone, Debug)]
 #[serde(untagged)]
 pub(crate) enum StandardEventConfig<T> {
-    Level(EventLevel),
+    Level(EventLevelConfig),
     Conditional {
-        level: EventLevel,
+        level: EventLevelConfig,
         condition: Condition<T>,
     },
 }
 
-#[derive(Debug, Clone)]
-pub(crate) enum StandardEvent<T> {
-    Level(EventLevel),
-    Conditional {
-        level: EventLevel,
-        condition: Arc<Mutex<Condition<T>>>,
-    },
-}
-
-impl<T> From<StandardEventConfig<T>> for StandardEvent<T> {
-    fn from(value: StandardEventConfig<T>) -> Self {
-        match value {
-            StandardEventConfig::Level(level) => StandardEvent::Level(level),
-            StandardEventConfig::Conditional { level, condition } => StandardEvent::Conditional {
-                level,
-                condition: Arc::new(Mutex::new(condition)),
-            },
+impl<T: Selector> StandardEventConfig<T> {
+    fn validate(&self, restricted_stage: Option<Stage>) -> Result<(), String> {
+        if let Self::Conditional { condition, .. } = self {
+            condition.validate(restricted_stage)
+        } else {
+            Ok(())
         }
     }
 }
 
 impl<T> Default for StandardEventConfig<T> {
     fn default() -> Self {
-        Self::Level(EventLevel::default())
+        Self::Level(EventLevelConfig::default())
     }
 }
 
-impl<T> StandardEvent<T> {
-    pub(crate) fn level(&self) -> EventLevel {
-        match self {
-            Self::Level(level) => *level,
-            Self::Conditional { level, .. } => *level,
-        }
-    }
+#[derive(Debug)]
+pub(crate) struct StandardEvent<T> {
+    pub(crate) level: EventLevel,
+    pub(crate) condition: Condition<T>,
+}
 
-    pub(crate) fn condition(&self) -> Option<&Arc<Mutex<Condition<T>>>> {
-        match self {
-            Self::Level(_) => None,
-            Self::Conditional { condition, .. } => Some(condition),
+impl<T: Clone> StandardEvent<T> {
+    pub(crate) fn from_config(config: &StandardEventConfig<T>) -> Option<Self> {
+        match &config {
+            StandardEventConfig::Level(level) => EventLevel::from_config(level).map(|level| Self {
+                level,
+                condition: Condition::True,
```

Comment:
Make it so the condition is always present and true by default.
Since now we have condition executed without Mutex for the most part it means cost of this simplification is pretty low.

Reply:
Ok I get the answer of my previous question about conditions not being optional